### DISCUSSION TITLE
Parallel routing

### DISCUSSION THREAD

**Discussion by louislefebvre** [remix-run/remix#4618] [General]
Hello :smile: !
If I am not mistaken, it is not possible today to have more than one child route in a route.

I am working on a dashboard where I would like to have for each component (kind of widget?) an error boundary, a loader and an action. I have made a quick drawing below.

As a junior developer, as this does not seem possible today if I am not mistaken, I am wondering if my request is interesting, or if I need to rethink my entire integration, or if I even have understood Remix routing correctly.

![image](https://user-images.githubusercontent.com/9850177/202210010-6c525013-d902-4325-886c-59e9b8983a84.png)
![image](https://user-images.githubusercontent.com/9850177/202210034-155387f6-5788-4742-9aee-3e0df48c2731.png)

Thanks!

**Comment by kiliman** [comment]
What you're asking for (each widget being a separate route) would be done with `<iframe>`, essentially each widget would be it's own page hosted inside the main page.

However, if you want a single page with multiple widgets that have their own loaders and ErrorBoundary, etc., Remix doesn't support it natively, but I do have a PoC that does exactly that. https://rmx.fyi/compdata

**Reply by louislefebvre** [reply]
Thanks! I see my confusion here where I mixed up child routes and components having their own loaders and ErrorBoundary.

May I ask what are the tradeoffs for having a route per widget and then embedding each route in an `iframe`?  If I am not mistaken, that seems to be easier than exposing components' loaders?

Moreover, you are saying that it is not a supported natively; is there any specific reason? Or should I re-think about my implementation/design?

**Reply by kiliman** [reply]
I mean that the Remix compiler doesn't look at components for loaders, actions, etc. It's still focused on routes only. So my PoC requires you to manually import these component loaders and call them directly.

**Comment by sergiodxa** [comment]
What I do to have this is to combine the [`handle` export](https://remix.run/docs/en/v1/api/conventions#handle) with [useMatches](https://remix.run/docs/en/v1/api/remix#usematches) to define a component to be rendered by the parent route layout along the Outlet, basically creating a second Outlet to be used by the layout.

**Reply by louislefebvre** [reply]
Thanks! May I ask if there is any example out there for `handle` usage? I will look into already existing examples.
